checks/evaluation/permissions/GitHubWorkflowPermissionsTopNoWrite.yml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+short: Checks that GitHub workflows do not have default write permissions
+desc: This rule checks that GitHub workflows do not have default write permissions
+motivation: >
+  If no permissions are declared, a workflow's GitHub token's permissions default to write for all scopes.
+  This include write permissions to push to the repository, to read encrypted secrets, etc.
+  For more information, see https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token. 
+implementation: >
+  The rule is implemented by checking whether the `permissions` keyword is defined at the top of the workflow,
+  and that no write permissions are given.
+risk: High
+remediation:
+  effort: Low
+  text:
+    - Visit https://app.stepsecurity.io/secureworkflow/${{ repo }}/${{ workflow }}/${{ branch }}?enable=permissions
+    - Tick the 'Restrict permissions for GITHUB_TOKEN'
+    - Untick other options
+  markdown:
+    - Visit [https://app.stepsecurity.io/secureworkflow](https://app.stepsecurity.io/secureworkflow/${{ repo }}/${{ workflow }}/${{ branch }}?enable=permissions).
```

Comment:
I want to suggest adding in the remediation text that to remediate multiple issues across different files in the repo, visit https://app.stepsecurity.io/securerepo. That way, various issues (token permissions, pinning dependencies, Dependabot config, adding CodeQL) can be fixed in a single pull request. The PR is created using a fork, and no App is needed to be installed. Many users already use this option but have to discover it themselves. Examples of PRs can be found [here](https://github.com/step-security/secure-workflows).

Reply:
Thanks for the suggestion. I'll try to add this into the text